ui/coordinateTrainer/src/ctrl.ts

```diff
@@ -147,11 +200,11 @@ export default class CoordinateTrainerCtrl {
   };
 
   private tick = () => {
-    const timeSpent = Math.min(DURATION, new Date().getTime() - +this.timeAtStart);
-    this.timeLeft = DURATION - timeSpent;
+    const timeSpent = Math.min(this.duration, new Date().getTime() - +this.timeAtStart);
+    this.timeLeft = this.duration - timeSpent;
     this.redraw();
 
-    if (this.timeLeft > 0) setTimeout(this.tick, TICK_DELAY);
+    if (this.timeLeft > 0 || this.timeDisabled) setTimeout(this.tick, TICK_DELAY);
```

Comment:
Not sure that we should be calling `this.tick` at all if time is disabled? Notice that tick calls redraw, so if we can avoid all these extra calls to redraw, that would be ideal.